config/stored_requests.go

```diff
@@ -11,8 +10,20 @@ import (
 	"github.com/golang/glog"
 )
 
+// DataType constants
+type DataType string
+
+const (
+	RequestDataType    DataType = "Request"
+	CategoryDataType   DataType = "Category"
+	VideoDataType      DataType = "Video"
+	AmpRequestDataType DataType = "Amp Request"
+)
+
 // StoredRequests configures the backend used to store requests on the server.
 type StoredRequests struct {
+	// DataType is a tag pushed from upstream indicating the type of object fetched here
+	DataType DataType
```

Comment:
There seems to be a trend in the feedback here about the placement of the DataTypes.

The `StoredRequests` and `StoredRequestsSlim` are config structs which the host can control. It seems strange IMHO to give the host the option of configuring the `DataType` field, so this doesn't seem like the opportune spot.

I recommend changing the signature of `CreateStoredRequests` to accept a `DataType` and pass it down as hardcoded values in `NewStoredRequests`. I think it would like nice paired with the config source:

```
fetcher1, shutdown1 := CreateStoredRequests(&slimAuction, RequestDataType, metricsEngine, client, router, &dbc)
fetcher2, shutdown2 := CreateStoredRequests(&slimAmp, metricsEngine, AmpRequestDataType , client, router, &dbc)
fetcher3, shutdown3 := CreateStoredRequests(&cfg.CategoryMapping, CategoryDataType, metricsEngine, client, router, &dbc)
fetcher4, shutdown4 := CreateStoredRequests(&cfg.StoredVideo, VideoDataType, metricsEngine, client, router, &dbc)
```

... or change the order fo have the `DataType` as the first argument if you find it reads better.

Reply:
I see DataType as metadata allowing any function passed the config to tell what kind it is. Passing the type separately gives us a pair of parameters meant to always be passed together, an indication to me that they should be fused.

I would have preferred to make it private so it is not exposed at all, but it needs to be readable so this implementation looked like a decent compromise.

The field itself lacks an annotation and is hardcoded, so it is not under the host control. 
It is similar to `BlackListedAppMap` or `BlackListedAcctMap`: https://github.com/prebid/prebid-server/blob/master/config/config.go#L61

I can change it to a separate parameter if that is still preferred.

Reply:
That's a good point. I can appreciate representing the pair of config and data type as a struct instead of always passing them together separately. Making the field private would be better IMHO, but as you've pointed out that won't work between packages.

The annotation on the config isn't actually required for serialization. So while I don't think anyone would specify a yaml config like:
```
stored_requests:
  datatype: "foo"
```
.. it would actually work, and that doesn't give me warm feelings.

What do you think of wrapping the `StoredRequests` in another struct which includes extra data, such as the type. Something like this (but with a hopefully better name):
```
type StoredRequestsWithMetadata struct {
  StoredRequests
  DataType DataType
}
```